orchestration/dagster_orchestration/hca_manage/verify_release_manifest.py

```diff
@@ -0,0 +1,128 @@
+                    AND state = 'succeeded'
+                    AND (sf.file_id IS NOT NULL OR
+                        af.file_id IS NOT NULL OR
+                        rf.file_id IS NOT NULL OR
+                        supf.file_id IS NOT NULL OR
+                        imgf.file_id IS NOT NULL)
+                )
+                SELECT REGEXP_EXTRACT(source_name, '(gs://.*)/data/.*$') AS bucket, count(*) as cnt
+                FROM base GROUP by bucket;
+                """
+    query_job = client.query(query)
+
+    # hydrate the rows
+    return [row for row in query_job]
+
+
+def parse_manifest_file(manifest_file):
+    with open(manifest_file) as manifest:
+        # some of the staging areas submitted via the form need slight cleanup
+        return [area.rstrip('\n/') for area in manifest]
+
+
+def verify(start_date, manifest_file, gs_project, bq_project, dataset):
+    logging.info("Parsing manifest and inspecting staging areas...")
+    creds = hca_google.get_credentials()
+    storage_client = storage.Client(project=gs_project, credentials=creds)
+    staging_areas = parse_manifest_file(manifest_file)
+    expected_load_totals = get_expected_load_totals(storage_client, staging_areas)
+
+    logging.info("Inspecting load history in bigquery...")
+    load_history = get_load_history(bq_project, dataset, start_date)
+    tdr_load_totals = {
+        load_history_row[0]: load_history_row[1]
+        for load_history_row in load_history
+    }
+
+    for unexpected_area in tdr_load_totals.keys() - expected_load_totals.keys():
+        logging.warning(f"⚠️ {unexpected_area} not in manifest but was imported")
+
+    success = True
+    total_loaded_count = 0
+    total_expected_count = 0
+    for area, expected_count in expected_load_totals.items():
+        total_expected_count += expected_count
+        if area in tdr_load_totals:
+            total_loaded_count += tdr_load_totals[area]
+            expected_count = expected_load_totals[area]
+            if expected_count != tdr_load_totals[area]:
+                logging.error(f"❌ Mismatched file count: staging_area = {area}, "
+                              f"imported = {tdr_load_totals[area]}, expected = {expected_count}")
+                success = False
+            else:
+                logging.info(
+                    f"✅ staging_area = {area}, imported = {tdr_load_totals[area]}, expected = {expected_count}")
+        else:
+            logging.error(f"❌ {area} has not been imported")
+            success = False
+    logging.info('-' * 80)
+    logging.info(
+        f"Total staging areas = {len(expected_load_totals.keys())}, total files staged = {total_expected_count}, total files loaded = {total_loaded_count}")
+    return success
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-s", "--start-date", required=True)
+    parser.add_argument("-f", "--manifest-file", required=True)
```

Comment:
Reading through the rest of the code, is manifest-file going to be the filename of the local file?

Reply:
Yes, this is a local csv extracted from the manifest spreadsheet. We can get fancy in a future iteration and connect to the google doc, but this works for now.